gr-blocks/lib/file_source_impl.cc

```diff
@@ -176,7 +176,7 @@ void file_source_impl::open(const char* filename,
     // If length is not specified, use the remainder of the file. Check alignment at end.
     if (length_items == 0) {
         length_items = items_available;
-        if (file_size % d_itemsize) {
+        if ((file_size != INT64_MAX) && (file_size % d_itemsize)) {
```

Comment:
Personally, I would have preferred if you just used `d_seekable` instead of `file_size != INT64_MAX` here – it describes the intent better.